frontend/lib/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -482,12 +551,25 @@ function DataFrame({
         disableFullscreenMode={disableFullscreenMode}
         // Lock the toolbar in some specific situations:
         locked={
-          isRowSelected || isCellSelected || (isTouchDevice && isFocused)
+          (isRowSelected && !isRowSelectionActivated) ||
+          isCellSelected ||
+          (isTouchDevice && isFocused)
         }
         onExpand={expand}
         onCollapse={collapse}
         target={StyledResizableContainer}
       >
+        {((isRowSelectionActivated && isRowSelected) ||
+          (isColumnSelectionActivated && isColumnSelected)) && (
+          <ToolbarAction
+            label={"Clear selection"}
+            icon={Close}
+            onClick={() => {
+              clearSelection()
+              clearTooltip()
+            }}
+          />
+        )}
```

Comment:
suggestion: I would probably move this further up:

```typescript
const includeToolbarAction = (isRowSelectionActivated && isRowSelected) ||
          (isColumnSelectionActivated && isColumnSelected);
const toolbarActionElement = includeToolbarAction && (
          <ToolbarAction
            label={"Clear selection"}
            icon={Close}
            onClick={() => {
              clearSelection()
              clearTooltip()
            }}
          />
        );
```
and then here just `{toolbarActionElement}`.

The whole component is already massive and for readability I think we should break the parts up a little bit